sklearn/feature_selection/tests/test_sequential.py

```diff
@@ -145,3 +223,8 @@ def test_pipeline_support():
     pipe = make_pipeline(StandardScaler(), sfs)
     pipe.fit(X, y)
     pipe.transform(X)
+
+
+def test_raise_deprecation_warning():
+    with pytest.warns(FutureWarning):
+        SequentialFeatureSelector(LinearRegression())
```

Comment:
The warning should happen only when calling fit. This test should fail indeed.

Reply:
warning is called from `__init__`

https://github.com/scikit-learn/scikit-learn/pull/20145/files#diff-521200bea206f673d4e933a33304f606f951d97edd7999b4d6b019cca79a81a0R159-R169

Reply:
Right, so I badly reviewed :). You should move this warning to `fit`. We always do parameter checking at `fit`. `__init__` is only to set the parameter as attributes.

Reply:
thanks, I understand !
move warnings from __init__ to fit and fix warning description with https://github.com/scikit-learn/scikit-learn/pull/20145/commits/05ffbdaab84b8d18f246497e65b0b7d306d681ad